### PR TITLE
One more click™️: part 3 (LOGGING!)

### PR DESCRIPTION
**Pull request by keirog**
(I’ve only added tests for the logs I think we may rely on for alerts/dashboards)

### CODE REVIEW

File: app/v2/one-click-subscribe/index.js [hunk 1]
```diff
@@ -15,16 +15,31 @@ const buildSignInUrl = ({
...
+	if (!userUuid && attemptsSoFar >= signInMaxAttempts) {
+		logger('error', { event: 'REDIRECTION_LIMIT_REACHED' });
 		return res.redirect(302, `${pagesBaseUrl}/newsletters/${newsletterId}/session-error`);
 	}
 
-	return res.redirect(302, buildSignInUrl({
-		attemptsSoFar,
-		newsletterId
-	}));
+	if (!userUuid) {
```
Comment (keirog) [inline app/v2/one-click-subscribe/index.js]: I was gonna do this in the next PR but had to pop it in here for the logging to work

File: app/v2/one-click-subscribe/index.js [hunk 2]
```diff
@@ -15,16 +15,31 @@ const buildSignInUrl = ({
...
+
+		logger('info', {
+			event: 'REDIRECTION_TO_SIGN_IN',
+			...details
+		});
+
+		return res.redirect(302, buildSignInUrl(details));
+	}
+
+	next();
```
Comment (keirog) [inline app/v2/one-click-subscribe/index.js]: ^ Just a temporary measure until we have the database call here

File: tests/integration/app/v2/one-click-subscribe/index.test.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 const request = require('supertest');
 const nock = require('nock');
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
```
Comment (AbdullahAli) [inline tests/integration/app/v2/one-click-subscribe/index.test.js]: I think we can avoid creating a sandbox, sinon now has a default sandbox: https://sinonjs.org/releases/latest/sandbox/#default-sandbox

File: tests/integration/app/v2/one-click-subscribe/index.test.js [hunk 4]
```diff
@@ -104,6 +108,16 @@ describe('One-click subscribe (v2)', () => {
 				.expect('Cache-Control', 'max-age=0, no-cache, must-revalidate')
 				.expect('Surrogate-Control', 'max-age=0, no-cache, must-revalidate');
 		});
+
+		it('calls logger.info', () => {
+			const loggerInfo = sandbox.spy(logger, 'info');
```
Comment (AbdullahAli) [inline tests/integration/app/v2/one-click-subscribe/index.test.js]: Super minor, but are you able to append `Spy` to the variable name to make it clearer when doing the expectations below 👍

File: tests/integration/app/v2/one-click-subscribe/index.test.js [hunk 5]
```diff
@@ -158,5 +172,40 @@ describe('One-click subscribe (v2)', () => {
 				.expect(302)
 				.expect('Location', `${pagesBaseUrl}/newsletters/${fakeNewsletterId}/unknown-error`);
 		});
+
+		it('calls logger.error if the Membership call fails', () => {
+			const loggerError = sandbox.spy(logger, 'error');
```
Comment (AbdullahAli) [inline tests/integration/app/v2/one-click-subscribe/index.test.js]: Similar to above, re: appending `Spy`

File: tests/integration/app/v2/one-click-subscribe/index.test.js (hunk 3 above)
Comment (keirog) [reply tests/integration/app/v2/one-click-subscribe/index.test.js]: Oh cool, ta!

File: tests/integration/app/v2/one-click-subscribe/index.test.js (hunk 4 above)
Comment (keirog) [reply tests/integration/app/v2/one-click-subscribe/index.test.js]: Of course, good shout

**Pull request merged**